src/pages/Admin/Venue/DetailsForm.tsx

```diff
@@ -181,41 +182,38 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
             },
             user
           );
-
           //@debt Create separate function that updates the userStatuses separately by venue id.
-          if (
-            sovereignVenueId &&
-            sovereignVenue &&
-            sovereignVenueId !== venueId
-          )
+          if (sovereignVenueId && sovereignVenueId !== venueId) {
+            const {
+              sovereignVenue: sovereignVenueData,
+            } = await fetchSovereignVenue(sovereignVenueId);
+
             await updateVenue(
               {
-                id: sovereignVenueId,
-                name: sovereignVenue.name,
+                id: sovereignVenueData.id,
+                name: sovereignVenueData.name,
                 subtitle:
-                  sovereignVenue.config?.landingPageConfig.subtitle ?? "",
+                  sovereignVenueData.config?.landingPageConfig.subtitle ?? "",
                 description:
-                  sovereignVenue.config?.landingPageConfig.description ?? "",
-                adultContent: sovereignVenue.adultContent ?? false,
-                profile_questions: sovereignVenue.profile_questions,
+                  sovereignVenueData.config?.landingPageConfig.description ??
+                  "",
+                adultContent: sovereignVenueData.adultContent ?? false,
+                profile_questions: sovereignVenueData.profile_questions,
                 code_of_conduct_questions:
-                  sovereignVenue.code_of_conduct_questions,
-                userStatuses,
-                showUserStatus: showUserStatuses,
-                template: sovereignVenue.template,
+                  sovereignVenueData.code_of_conduct_questions,
+                userStatuses: sovereignVenueData.userStatuses,
+                showUserStatus: sovereignVenueData.showUserStatus,
+                template: sovereignVenueData.template,
               },
               user
             ).then(() => {
-              if (sovereignVenue) {
-                dispatch(
-                  setSovereignVenue({
-                    ...sovereignVenue,
-                    userStatuses,
-                    showUserStatus: showUserStatuses,
-                  })
-                );
-              }
+              dispatch(
+                setSovereignVenue({
+                  ...sovereignVenueData,
+                })
+              );
             });
```

Comment:
LGTM other than it could use a bit of declutter to clarify the actual data being merged. Examples:

```ts
            const { sovereignVenue } = await fetchSovereignVenue(
              sovereignVenueId
            );

            const updateData: WithId<VenueInput> = {
              adultContent: false,
              ...pick(sovereignVenue, [
                "id",
                "name",
                "adultContent",
                "profile_questions",
                "code_of_conduct_questions",
                "userStatuses",
                "showUserStatus",
                "template",
              ]),
              subtitle: "",
              description: "",
              ...pick(sovereignVenue.config?.landingPageConfig, [
                "subtitle",
                "description",
              ]),
            };

            await updateVenue(updateData, user).then(() =>
              dispatch(setSovereignVenue(sovereignVenue))
            );
          }
``` 

```ts
            const { sovereignVenue } = await fetchSovereignVenue(
              sovereignVenueId
            );

            const {
              id,
              name,
              adultContent = false,
              profile_questions,
              code_of_conduct_questions,
              userStatuses,
              showUserStatus,
              template,
            } = sovereignVenue;

            const { subtitle = "", description = "" } =
              sovereignVenue.config?.landingPageConfig ?? {};

            await updateVenue(
              {
                id,
                name,
                subtitle,
                description,
                adultContent,
                profile_questions,
                code_of_conduct_questions,
                userStatuses,
                showUserStatus,
                template,
              },
              user
            ).then(() => dispatch(setSovereignVenue(sovereignVenue)));
```

Reply:
fixed in f8d98f7